File: website/src/__generated/services/ChildReplyController.ts
import type { Executor } from '../';
import type { ReplyDto } from '../model/dto';
import type { Page } from '../model/static';

export class ChildReplyController {
    
    constructor(private executor: Executor) {}
    
    async findChildrenReplies(options: ChildReplyControllerOptions['findChildrenReplies']): Promise<
        Page<ReplyDto['DEFAULT']>
    > {
        let _uri = '/categories/forum/thread/replies/';
        _uri += encodeURIComponent(options.replyId);
        _uri += '/children/';
        let _separator = _uri.indexOf('?') === -1 ? '?' : '&';
        let _value: any = undefined;
        _value = options.page;
        if (_value !== undefined && _value !== null) {
            _uri += _separator
            _uri += 'page='
            _uri += encodeURIComponent(_value);
            _separator = '&';
        }
        _value = options.size;
        if (_value !== undefined && _value !== null) {
            _uri += _separator
            _uri += 'size='
            _uri += encodeURIComponent(_value);
            _separator = '&';
        }
        return (await this.executor({uri: _uri, method: 'GET'})) as Page<ReplyDto['DEFAULT']>
    }
}

export type ChildReplyControllerOptions = {
    'findChildrenReplies': {
        readonly page?: number, 
        readonly size?: number, 
        readonly replyId: string
    }
}

File: server/src/main/kotlin/cn/enaium/zensquare/controller/category/forum/thread/reply/child/ChildReplyController.kt
/*
 * ZenSquare is an opensource forums
 *
 * Copyright (C) 2023  Enaium
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published
 * by the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package cn.enaium.zensquare.controller.category.forum.thread.reply.child

import cn.enaium.zensquare.model.entity.Reply
import cn.enaium.zensquare.repository.ReplyRepository
import org.springframework.data.domain.Page
import org.springframework.data.domain.PageRequest
import org.springframework.web.bind.annotation.*
import java.util.*

/**
 * child reply controller
 *
 * @author Enaium
 */
@RequestMapping
@RestController
class ChildReplyController(val replyRepository: ReplyRepository) {
    /**
     * Find all child replies
     *
     * @param page page
     * @param size size
     * @param replyId reply id
     * @return Page<Reply>
     */
    @GetMapping("/categories/forum/thread/replies/{replyId}/children/")
    fun findChildrenReplies(
        @RequestParam(defaultValue = "0") page: Int = 0,
        @RequestParam(defaultValue = "10") size: Int = 10,
        @PathVariable replyId: UUID
    ): Page<Reply> {
        return replyRepository.findAllByParentId(PageRequest.of(page, size), replyId)
    }
}